### ISSUE TITLE
Front End: Update VAMC appointment information in Service Location

### ISSUE THREAD

**Issue by xiongjaneg** [department-of-veterans-affairs/va.gov-cms#16144]
## User Story or Problem Statement
VAMC Facility Service appointment information will be migrated into updated Service Location in #15559. This aligns with the VBA appointments information being in Service Location. This will result in changes to how this information is displayed in accordions.

- [Mural of the content model](https://app.mural.co/t/departmentofveteransaffairs9999/m/vagov6717/1677277602711/74cecb64b6eeb9558aa868240cd726b54fcb93a7?sender=25a80b92-ba60-455b-8532-e5850ea9bc2a)
- [VBA Service accordions in Sketch](https://www.sketch.com/s/891d33ae-152d-471c-ab5d-9aedf89cf6ff/p/DC11E46F-AF92-4DFD-AC0B-BEB9A9DD3C9A/canvas)
- [VAMC Service accordianns in Sketch](https://www.sketch.com/s/8bda90fd-99e8-4761-9210-3a24282eb811)

Changes to Drupal for service locations is still are still in [this PR](https://github.com/department-of-veterans-affairs/va.gov-cms/pull/15622) awaiting change management to VAMC

This needs to merge a day after #15559 makes it to prod to prevent editorial confusion.

- Needs to account for Walk-ins accepted going away and Service Delivery Options being added.
- Review [Sketch](https://www.sketch.com/s/8bda90fd-99e8-4761-9210-3a24282eb811) for design updates (borders to separate info, order of information).

- service location integration https://pr15622-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/

## Acceptance Criteria
- [x] Timing is coordinated with Drupal engineers
- [x] Query is updated to remove these fields from VAMC Facility Service (health_care_local_health_service) 
   - [x]  field_hservice_appt_intro_select
   - [x]  field_hservice_appt_leadin
   - [x] field_online_scheduling_availabl
   - [x] field_walk_ins_accepted
   - [x]  field_phone_numbers_paragraph
- [x] Query is updated to add the fields from the Service Location paragraph.
   - [x] Not sure what fields have already been added.  This will give you the [list of all the fields in the service location paragraph](https://pr15622-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/admin/reports/content-model/fields?field_name=&field_type=&entity_type=paragraph&bundle=service_location&label=)
- [x] Template is updated to remove the deleted fields and add the new fields.
  - [x] Walk-ins accepted is no longer showing.
  - [x] Service Delivery options are showing.
  - [x] For Appointments intro text, if Customize text is selected but no text is entered, nothing is shown (acts like Remove text was selected)
- [x] Accordion elements are confirmed
- [x] Merge the VBA and VAMC Facility and Regional Service Location paragraph type liquid templates.
- [ ] This ticket need to go into their own integration branch.
- [ ] Testing is completed as needed
- [ ] Design is reviewed (tag in PR)
- [ ] Accessibility is reviewed (tag in PR)

## Team
Please check the team(s) that will do this work.

- [ ] CMS Team
- [ ] Public Websites
- [x] Facilities
- [ ] Accelerated Publishing

**Comment by xiongjaneg** [comment]
Already have a button styling ticket.

Queries have some overlap, so keep in one ticket and keep the commits separate.

It doesn't have to be one PR but does have to be one integration branch.

**Comment by mmiddaugh** [comment]
@xiongjaneg I'd like to see us implement the full design with VAMC health services because we know it is challenging to scan and parse the information as separate service locations in the current state.

But let's talk through this issue during refinement 
- to understand comparative levels of effort to 1) implement only what is necessary to accommodate data changes to VAMC health services vs 2) also implementing the visual parts of Jordan's design for VAMC health services
- understand how the work could/should be broken down further (i.e. #15601)

**Comment by swirtSJW** [comment]
Data will be coming from this tugboat instance https://pr15622-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/

**Comment by eselkin** [comment]
Just putting this here in case I lose the location of the Health Care Local Facility GraphQL
`src/site/stages/build/drupal/graphql/healthCareLocalFacilityPage.graphql.js`

**Comment by eselkin** [comment]
@swirtSJW
Having trouble finding where the service location paragraph type is housed.

I see on NodeHealthCareLocalFacility there's a LocationService which can be a ParagraphServiceLocation, but is that the thing we're looking for?

Or is it connected some other way? I'm seeing empty ParagraphServiceLocaiton (no matching elements) for any of the Local Facilities I've searched through.

**Comment by eselkin** [comment]
@xiongjaneg
Are we mimicking the design from the VBA accordions or keeping the design like the service locations paragraphs that exist already in VAMCs? Are we adding the icons for the walkins etc? Are we using the components like the online scheduling active links etc?

**Comment by xiongjaneg** [comment]
@eselkin Apologies, I forgot to link to Jordan's recently completed [design for VAMC service location](https://www.sketch.com/s/8bda90fd-99e8-4761-9210-3a24282eb811)

**Comment by xiongjaneg** [comment]
From Steve: These should all be the same service location partial / component. That would cover VBA, non clinical services and VAMC service.

**Comment by xiongjaneg** [comment]
From Eli: There is some test in VAMC that doesn't render the entire paragraph, which is different from VBA. We can unify these but need to look at the conditionals that are currently used in VAMC service location paragraph and potentially copy in the VBA paragraph. We just broke out the VBA service location paragraphs so we can look at merging the two service location templates.

We don't have sample VBA data so we weren't exactly sure what was generated and what it would look like. We will need to generate that data for the service location migration.

**Comment by xiongjaneg** [comment]
@xiongjaneg stub ticket for other work outside of the scope of the ACs in this ticket.

**Comment by davidmpickett** [comment]
@eselkin I added some notes on your PR, but not sure what parts can be determined there vs what need to be split off into other tickets.

**Comment by xiongjaneg** [comment]
In progress. The challenge is that it's using a service location paragraph liquid template shared between VBA and VAMC. The problem is that they don't both align 100% on what they have and the spacing as a result of this between the two of them messes up the appearance.

**Comment by jilladams** [comment]
@eselkin can you verify status for end of sprint,as I missed the Service Location reckoning meeting yesterday? From PR it looks like for end of sprint: have addressed many pieces of design feedback, and I saw one remaining open question for @davidmpickett , but not sure if that's everything. 

The one open question for Dave looked like: 

Q:
> This isn't good, but I don't know how to fix it
> If an editor has entered building name/number or wing, floor or room number, but not Service location address this information now will be floating around without context. Vs on Prod, this information would default to falling under the generic Contact Info header

Proposed Answer: Right now if there's no clinic name but there is subsequent data, we could default to something like:
Location Info
In the header space for that section.

Eli can you verify if I have that right as far as status / what remains to get sorted in the PR? (I missed the serv

**Comment by davidmpickett** [comment]
> Q:
> 
> This isn't good, but I don't know how to fix it
> If an editor has entered building name/number or wing, floor or room number, but not Service location address this information now will be floating around without context. Vs on Prod, this information would default to falling under the generic Contact Info header
> 
> Proposed Answer: Right now if there's no clinic name but there is subsequent data, we could default to something like:
> Location Info
> In the header space for that section.

Might make sense to migrate this open question to @thejordanwood's new ticket for reviewing the design of non-clinical services.
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17342

Or somewhere else in the new epic
https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17350

**Comment by laflannery** [comment]
@eselkin all the headings levels look good, I have no more comments or changes, this is approved by me.

Also, I'm adding all of the heading documentation I can possible add for sanity and for future reference and for all the things -

I reviewed the below pages and confirmed they all have proper headings levels regarding this new service location/situation display:
- Billing and Insurance
  - [Fayetteville Arkansas](https://web-pr15622-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/fayetteville-arkansas-health-care/billing-and-insurance/)
  - [Washington DC](https://web-pr15622-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/washington-dc-health-care/billing-and-insurance/)
  - Screenshot of heading levels:
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/94919339-27a7-4910-9bb5-ad9c347c5fef)
- Medical Records
  - [Fayetteville Arkansas](https://web-pr15622-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/fayetteville-arkansas-health-care/medical-records-office/)
  - [Washington DC](https://web-pr15622-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/washington-dc-health-care/medical-records-office/)
  - Screenshot of heading levels:
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/d9015065-00fc-47e5-a9d5-14585203c4f7)
- Register for Care
  - [Fayetteville Arkansas](https://web-pr15622-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/fayetteville-arkansas-health-care/register-for-care/) - Does not actually have any service locations so this was not helpful for this case
  - [Washington DC](https://web-pr15622-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/washington-dc-health-care/register-for-care/)
  - Screenshot of heading levels:
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/2f126bc6-a556-4bc0-81f3-364c2441bbe6)
- VAMC Clinical Services
  - [Fayetteville Arkansas VA Medical Center - Audiology and Speech](https://web-pr15622-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/fayetteville-arkansas-health-care/locations/fayetteville-va-medical-center/#audiology-and-speech)
  - [Washington VA Medical Center  - Audiology and Speech](https://web-pr15622-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/washington-dc-health-care/locations/washington-va-medical-center/#audiology-and-speech)
  - Screenshot of heading levels:
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/d6d7d222-fa23-4de1-bc84-d36c1d70e6f4)
- VBA
  - [Cleveland VA Regional Benefit Office](https://web-pr15622-xptmfsnjoodldn85nzgkrgljqtfbz24b.ci.cms.va.gov/cleveland-va-regional-benefit-office/locations/cleveland-va-regional-benefit-office/)
  - Screenshot of heading levels: (FYI The accordion is H3, my other extension wasn't working and this particular extension doesn't see the shadow DOM so I had to write it in)
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/14a23356-09ce-43a8-8111-eb1ba9a04940)

**Comment by jilladams** [comment]
@eselkin also: In sprint planning, we opted to take the DaveP Location Info question into #17342, so you can exclude that from your current PR. Which _might_ mean that addressing target branch is the last blocker to getting this out the door for you?

**Comment by jilladams** [comment]
From planning: we noted this PR _is_ the integration branch. (🤦‍♀️ ) So: work is complete now, ticket is pending integration when Drupal work is ready to merge.
